### ISSUE TITLE
UnsupportedOperationException on Samsung Galaxy S3 (GT-9300)

### ISSUE THREAD

**Issue by ivanshatsky** [cvzi/ScreenshotTile#165]
On my Samsung Galaxy S3 (GT-9300), when I tried to take a screenshot, I've got the following error:

```
Failed to take screenshot, please try again
Could not acquire image
UnsupportedOperationException
This device is not supported.
```

The phone OS is LineageOS 14.1 (Android 7.1.2).
The app version is 1.17.0 (66), downloaded from the F-Droid site.
Is there any chances to make the Screenshot Tile work on this phone? Maybe installing some third party libraries, etc?
Thanks in advance!

**Comment by cvzi** [comment]
I have created three apks with different settings, that you can test. Maybe one of these works:
[apks.zip](https://github.com/cvzi/ScreenshotTile/files/8882010/apks.zip)

You need to uninstall the app before installing one of the apk and then uninstall again before installing the next apk.

If you now how, you can also try to find the full error message in the logs with adb. If you search for `UnsupportedOperationException` in the logs, you should be able to find a longer error message.
Here is an explanation on how to find logs with adb: https://thedarksource.com/adb-logcat-command-to-capture-debug-logs-from-android-device/

**Comment by ivanshatsky** [comment]
Thank you for your assistance! I tried all the three apks. Unfortunately, none of them works for me. However error messages became somewhat more informative:
- app-release-420.apk:
```
Failed to take screenshot, please try again
YUV_420_888:UnsupportedOperationException
The producer output buffer format 0x5 doesn't match the ImageReader's configured buffer format 0x23.
```
- app-release-565.apk:
```
Failed to take screenshot, please try again
RGB_565:UnsupportedOperationException
The producer output buffer format 0x5 doesn't match the ImageReader's configured buffer format 0x4.
```
- app-release-8888.apk:
```
Failed to take screenshot, please try again
RGBX_8888:UnsupportedOperationException
The producer output buffer format 0x5 doesn't match the ImageReader's configured buffer format 0x2.
```
Does it help to track down the problem in some way? Do you still need the full error log messages? I did't install the ADB driver yet, however if it can provide some more helpful information you needed, I think I can do it too.

**Comment by cvzi** [comment]
Yes, that helps. 

No need for the adb log.


El 11 de junio de 2022 15:22:53 CEST, Ivan Shatsky ***@***.***> escribió:
>Thank you for your assistance! I tried all the three apks. Unfortunately, none of them works for me. However error messages became somewhat more informative:
>- app-release-420.apk:
>```
>Failed to take screenshot, please try again
>YUV_420_888:UnsupportedOperationException
>The producer output buffer format 0x5 doesn't match the ImageReader's configured buffer format 0x23.
>```
>- app-release-565.apk:
>```
>Failed to take screenshot, please try again
>RGB_565:UnsupportedOperationException
>The producer output buffer format 0x5 doesn't match the ImageReader's configured buffer format 0x4.
>```
>- app-release-8888.apk:
>```
>Failed to take screenshot, please try again
>RGBX_8888:UnsupportedOperationException
>The producer output buffer format 0x5 doesn't match the ImageReader's configured buffer format 0x2.
>```
>Does it help to track down the problem in some way? Do you still need the full error log messages? I did't install the ADB driver yet, however if it can provide some more helpful information you needed, I think I can do it too.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/cvzi/ScreenshotTile/issues/165#issuecomment-1152927513
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Enviado desde mi dispositivo Android con K-9 Mail. Por favor, disculpa mi brevedad.

**Comment by cvzi** [comment]
It seems like there is a bug in LineageOS on Samsung devices, maybe in the graphics driver. It returns this `format 0x5`, but that format is not accepted by the Android SDK.

I have created two more apks to test:
[apks2.zip](https://github.com/cvzi/ScreenshotTile/files/8884779/apks2.zip)

**Comment by ivanshatsky** [comment]
Unfortunally, no luck with those two either.
- app-release-jpeg.apk
```
r=256, d=1
UnsupportedOperationException
The producer output buffer format 0x5 doesn't match the ImageReader's configured buffer format 0x21.
```
- app-release-0x5.apk
```
UnsupportedOperationException
Invalid format specified 5
```
I got you point about the bug, most likely in the Samsung video driver shipped with LineageOS 14.1. I think I should try to update the OS/driver instead, the app itself doesn't behave incorrectly. I'd write an additional instructions here for anyone faced the same issue if I'd succeed. Meanwhile I think you can close this one. Anyway, thanks for trying to help!
